### PR TITLE
fix doc links

### PR DESCRIPTION
**Pull request by benisraeldan**
fixes links in docs

### CODE REVIEW

File: docs/source/user-guide/dataset_object.rst [hunk 1]
```diff
@@ -163,5 +163,5 @@ will run on the columns to determine which are classified as categorical:
    - If (`number of unique values in column` <= ``max_categories`)  
      **AND** ((the ratio between the `number of unique values` and the `number of samples`) < `max_categorical_ratio`)
   
-Check the API Reference for :doc:`infer_categorical_features </api/utils/generated/deepchecks.utils.features.infer_categorical_features>` 
+Check the API Reference for :doc:`infer_categorical_features </api/generated/deepchecks.utils.features.infer_categorical_features>`
```
Comment (ItayGabbay) [inline docs/source/user-guide/dataset_object.rst:166]: @yromanyshyn is this the proper way to do this? (After the extension you added?)

Comment (shir22) [reply docs/source/user-guide/dataset_object.rst:166]: waiting for your answer also for #643

Comment (yromanyshyn) [reply docs/source/user-guide/dataset_object.rst:166]: yes, it looks like a proper way
the problem is the same as with #688
I am working over it

**Review by shir22 — APPROVED** [review]

File: docs/source/user-guide/dataset_object.rst (hunk 1 above)
Comment (yromanyshyn) [reply docs/source/user-guide/dataset_object.rst:166]: I merged #688 please pull and try

**Pull request merged**